### PR TITLE
feat(dbx-edge-data-protection): data protection included

### PR DESCRIPTION
**Pull request by athielba**
## Description

1. Sync folders & file
2. Delete object permission
3. Refresh object

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/dropbox-edge/src/inngest/functions/organisations/remove-organisation.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ export const removeOrganisation = inngest.createFunction(
 
     await elba.connectionStatus.update({ hasError: true });
 
+    // Delete Shared Links(If exist)
+    await db.delete(sharedLinksTable).where(eq(organisationsTable.id, organisationId));
```
Comment (elbalexandre) [inline apps/dropbox-edge/src/inngest/functions/organisations/remove-organisation.ts:46]: This should be deleted in cascade

File: apps/dropbox-edge/src/database/schema.ts [hunk 2]
```diff
@@ -12,3 +12,22 @@ export const organisationsTable = pgTable('organisations', {
 });
 
 export type Organisation = InferSelectModel<typeof organisationsTable>;
+
+export const sharedLinksTable = pgTable(
+  'shared_links',
+  {
+    id: text('id').notNull(),
+    url: text('url').notNull(),
+    organisationId: uuid('organisation_id').notNull(),
```
Comment (elbalexandre) [inline apps/dropbox-edge/src/database/schema.ts:21]: This should be an FK of organisation with delete cascade on it

File: apps/dropbox/src/app/api/webhooks/elba/data-protection/delete-object-permissions/service.ts [hunk 3]
```diff
@@ -1,5 +1,25 @@
-import { DeleteObjectPermissions } from '@/connectors/types';
 import { inngest } from '@/inngest/client';
```
Comment (elbalexandre) [inline apps/dropbox/src/app/api/webhooks/elba/data-protection/delete-object-permissions/service.ts:1]: I guess we don't need thoses changes in this app

Comment (athielba) [reply apps/dropbox/src/app/api/webhooks/elba/data-protection/delete-object-permissions/service.ts:1]: This one is in the old dropbox integration

**Review by elbalexandre — APPROVED** [review]

**Pull request merged**